include/boost/geometry/strategies/is_convex/cartesian.hpp

```diff
@@ -0,0 +1,77 @@
+// Boost.Geometry
+
+// Copyright (c) 2021, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_IS_CONVEX_CARTESIAN_HPP
+#define BOOST_GEOMETRY_STRATEGIES_IS_CONVEX_CARTESIAN_HPP
+
+
+#include <boost/geometry/strategies/cartesian/point_in_point.hpp>
+#include <boost/geometry/strategies/cartesian/side_by_triangle.hpp>
+#include <boost/geometry/strategies/convex_hull/cartesian.hpp>
+#include <boost/geometry/strategies/detail.hpp>
+#include <boost/geometry/strategies/is_convex/services.hpp>
+#include <boost/geometry/util/type_traits.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategies { namespace is_convex
+{
+
+template <typename CalculationType = void>
+class cartesian : public strategies::convex_hull::cartesian<CalculationType>
+{
+public:
+    template <typename Geometry1, typename Geometry2>
+    static auto relate(Geometry1 const&, Geometry2 const&,
+                       std::enable_if_t
+                            <
+                                util::is_pointlike<Geometry1>::value
+                             && util::is_pointlike<Geometry2>::value
```

Comment:
This should probably be replaced by
`distance::detail::enable_if_pp_t<Geometry1, Geometry2>` ?

Reply:
But that is in `distance`, maybe it's more generic than that?

Reply:
The problem is that the naming correspond to the strategy, not geometries, e.g. enable_if_ps is true also for "segmental" geometries so two linestrings or two polygons. This is why I put it in `distance::detail`. But I agree that in general a utility like that could be useful. I'd only have to review all of the strategies and come up with a proper names. But I'd prefer to do this in a different PR.

Reply:
In another PR is fine.